pipeline.py

```diff
@@ -51,19 +59,26 @@ class CorpusManager:
     """
 
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        path_to_raws = Path(self.path_to_raw_txt_data)
+        dataset = list(path_to_raws.rglob('*_raw.txt'))
+
+        for file in dataset:
+            article_id = int(file.parts[-1].split('_')[0])
```

Comment:
you are too sure that there will be a number at the beginning of a file. the way you extract numbers is extremely not correct

Reply:
fixed